crates/bevy_gilrs/src/converter.rs

```diff
@@ -25,21 +25,22 @@ pub fn convert_button(button: gilrs::Button) -> Option<GamepadButtonType> {
         gilrs::Button::DPadDown => Some(GamepadButtonType::DPadDown),
         gilrs::Button::DPadLeft => Some(GamepadButtonType::DPadLeft),
         gilrs::Button::DPadRight => Some(GamepadButtonType::DPadRight),
-        gilrs::Button::Unknown => None,
+        gilrs::Button::Unknown => Some(GamepadButtonType::Other(code.into_u32() as u8)),
```

Comment:
The code format is dependent on the platform, and on some (macOS and window/wgi) it's bigger than a `u32`. Converting it again to a `u8` means that only windows/xinput and wasm won't lose data.

Reply:
Ah, fair 'nuff, yeah I see now that's what gilrs [does on mac](https://github.com/Arvamer/gilrs/blob/master/gilrs-core/src/platform/macos/gamepad.rs#L421).

Bevy's `Other` variant only stores a `u8`: https://github.com/bevyengine/bevy/blob/main/crates/bevy_input/src/gamepad.rs#L208 would it make sense to up that to a `u32`? That would increase the size of the enum, and would also require that users adjust what type they use for that (which admittedly would probably be a fairly simple refactor).

Reply:
I think we should bump it up for consistency: I don't think it makes sense to have an "other" variant without proper support.

Reply:
or don't use that `into_u32` function that makes no guarantee on its meaning and keep the platform dependent `EvCode`? It's impossible to do anything cross platform with this value, at least it would make it clear

Reply:
So, I tried storing a `gilrs::ev::Code` in the `Other` variant, but ran into two problems:
* `bevy_input` needs to depend directly on `gilrs`, and
* We need `gilrs::ev::Code` to implement `FromReflect` because `GamepadAxisType` implements `FromReflect`.

It also means that if someone were using serde to send gamepad bindings between platforms, which previously would work (and is cross-platform for every variant except `Other`), now it possibly won't deserialize properly. Which we probably want to stop them from doing that, maybe? Except that it _is_ valid to do that for every variant except `Other`.